src/main/java/org/jenkinsci/plugins/workflow/actions/ExecutorTaskInfoAction.java

```diff
@@ -0,0 +1,88 @@
+package org.jenkinsci.plugins.workflow.actions;
+
+import hudson.model.InvisibleAction;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * Records information for a {@code node} block.
+ */
+public class ExecutorTaskInfoAction extends InvisibleAction implements PersistentAction {
+    private static final long serialVersionUID = 1;
+
+    private String whyBlocked;
+    private boolean launched;
+    // Initialized at -1 for "not started yet".
+    private long whenStartedOrCancelled = -1L;
+
+    public ExecutorTaskInfoAction() {
+    }
+
+    public ExecutorTaskInfoAction(@Nonnull String whyBlocked) {
+        this.whyBlocked = whyBlocked;
+    }
+
+    public void setLaunched() {
+        // Because we're not blocked any more at this point!
+        this.whyBlocked = null;
+        this.whenStartedOrCancelled = System.currentTimeMillis();
+        this.launched = true;
+    }
+
+    public void setWhyBlocked(@Nonnull String whyBlocked) {
+        this.whyBlocked = whyBlocked;
+    }
+
+    @CheckForNull
+    public String getWhyBlocked() {
+        return whyBlocked;
+    }
+
+    public long getWhenStartedOrCancelled() {
+        return whenStartedOrCancelled;
+    }
+
+    public void cancelTask() {
+        this.whyBlocked = null;
+        this.whenStartedOrCancelled = System.currentTimeMillis();
+    }
+
+    public boolean isQueued() {
+        return whyBlocked != null && whenStartedOrCancelled == -1;
+    }
+
+    public boolean isLaunched() {
+        return launched;
+    }
+
+    public boolean isCancelled() {
+        return !launched && whyBlocked == null && whenStartedOrCancelled > -1;
+    }
+
+    public static boolean isNodeQueued(@Nonnull FlowNode node) {
```

Comment:
I mean that rather than booleans, have an enum for the states defined for queued tasks [here](https://github.com/jenkinsci/jenkins/blob/51c46c6cf22a57860c71c7d7236ae30f6baa6651/core/src/main/java/hudson/model/Queue.java#L145). 

And yeah I dunno why that never had an enum for Queue task state since a task should only be in one of the `ItemList`s -- this is a heck of a lot simpler than checking different conditionals